### ISSUE TITLE
Figure out how to open the papyri browser from the kernel.

### ISSUE THREAD

**Issue by Carreau** [Carreau/papyri-lab#2]
I see two possibilities:

 - when we display help with `?`, IPython sends a mimebundle with test and html, can we hook into a specific mimetype that can trigger the opening of the papyri browser. 
 - the inspector might already be able to to hook into that as it's content updates as users are selecting text. How does that work ? Can we reuse this mechanism ?

**Comment by Carreau** [comment]
@peytondmurray you are more familiar than me with JupyterLab, can you try to have a look? And maybe give a rough idea ?

**Comment by peytondmurray** [comment]
Yes, I'll take a look and see if there's an elegant way to do this.

**Comment by peytondmurray** [comment]
I've started looking into the inspector, as that seemed very close to what we already wanted to do.

The inspector works by creating an `InspectionHandler` associated with each notebook or console opened by a user. The `InspectionHandler` listens for signals from a `CodeEditor.IEditor`, which are emitted whenever the user

1. Changes the selection
2. Changes the code
3. Changes the selected cell

The `InspectionHandler` uses `CodeEditor` to find the position of the cursor, and an ["introspection" request](https://jupyter-client.readthedocs.io/en/latest/messaging.html#introspection) is sent to the kernel. It's up to the kernel what is returned in an introspection request, but for IPython it is evidently the docstring for the object in question.

To me, it seems like the most useful part of this approach is in finding the position of the cursor. Maybe then we can somehow translate that into an `ILocation` for papyri? This path seems somewhat complicated.

I'll keep digging into other ways that we can hook into `?`, there could be an easier way to do this via your first suggestion.

**Comment by peytondmurray** [comment]
Okay, I figured out how to do this: we need to spy on the kernel by connecting to the [`IKernelConnection.anyMessage`](https://jupyterlab.github.io/jupyterlab/modules/_services_src_index_._services_src_kernel_kernel_.ikernelconnection.html#anymessage) signal, which gets emitted any time the kernel sends/receives a message. With this, we can listen for code which is sent with a `?` or `??`, and open the papyri window automatically. I'll get right on implementing a proof of concept.
